src/components/Calculator.js

```diff
@@ -1,52 +1,48 @@
-import React from 'react';
-import '../index.css';
+import React, { Component } from 'react';
+import calculate from '../logic/calculate';
 
-export default class Calculator extends React.Component {
+const operators = ['÷', 'x', '-', '+', '='];
+const digits = ['AC', '+/-', '%', '7', '8', '9', '4', '5', '6', '1', '2', '3', '0', '.'];
+
+class Calculator extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      next: '',
+      total: '',
+      operation: '',
+    };
+  }
+
+  handleClick = (btn) => {
+    this.setState((state) => calculate(
+      {
+        next: state.next,
+        total: state.total,
+        operation: state.operation,
+      },
+      btn,
+    ));
   }
 
   render() {
+    const { total, next } = this.state;
     return (
-      <div className="calculator">
-        <div className="calculator__view__container">
-          <div className="calculator__view">
-            <h1 className="calculator__view__h1">0</h1>
-          </div>
-        </div>
-        <div className="calculator__rows">
-          <div className="calculator__row">
-            <button type="button" className="calculator__greyButton">AC</button>
-            <button type="button" className="calculator__greyButton">+/-</button>
-            <button type="button" className="calculator__greyButton">%</button>
-            <button type="button" className="calculator__orangeButton">÷</button>
+      <ul className="calculator-body">
+        <li className="calculator-screen">
+          {next || total || '0'}
+        </li>
+        <li className="calculator-keyboard">
+          <div className="calculator-digits">
+            {digits.map((btn) => <button onClick={() => this.handleClick(btn)} type="button" key={btn}>{ btn }</button>) }
           </div>
```

Comment:
- [ ] Your app throws an error on the console when I divide a number by 0. Kindly fix this issue by catching this error and throwing a message that the user cannot make this math operation. 

<img width="723" alt="image" src="https://user-images.githubusercontent.com/77190203/159099775-79419070-cb74-429e-afb7-0c0d0a8a9d5e.png">